src/json-api/json-api-hapi-plugin.ts

```diff
@@ -589,20 +640,69 @@ export class JsonApiHapiPlugin {
     return reply(this.getEntity('notification', api => api.getProjectNotificationConfigurations(projectId)));
   }
 
-  public async postNotificationConfigurationHandler(request: Hapi.Request, reply: Hapi.IReply) {
-    const config = request.payload.data.attributes;
-
-    config.teamId = config.teamId || null;
-    config.projectId = config.projectId || null;
+  private async authorizeNotificationConfiguration(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      reply(request.payload.data.attributes);
+    } catch (error) {
+      reply(Boom.badRequest());
+    }
+  }
 
-    if (!config.teamId && !config.projectId) {
-      throw Boom.badRequest('teamId or projectId should be defined');
+  private async authorizeTeamOrProjectAccess(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      const config = request.pre.filter;
```

Comment:
This pre function is used as the second one in a chain of two with the implicit contract that the initial pre function sets an object containing either the projectId or the teamId in the key `filter`. This key is defined in the route configuration and could be anything. 

I think that the method name is fine, as it doesn't care about the actual request.